spk/tt-rss/src/dsm-control.sh

```diff
@@ -10,15 +10,70 @@ WEB_DIR="/var/services/web"
 PATH="${INSTALL_DIR}/bin:${PATH}"
 BUILDNUMBER="$(/bin/get_key_value /etc.defaults/VERSION buildnumber)"
 USER="$([ "${BUILDNUMBER}" -ge "4418" ] && echo -n http || echo -n nobody)"
-PHP="$([ "${BUILDNUMBER}" -ge "7321" ] && echo -n /usr/local/bin/php56 || echo -n /usr/bin/php)"
+
 TTRSS="${WEB_DIR}/${PACKAGE}/update.php"
+HT_ACCESS_FILE="${WEB_DIR}/${PACKAGE}/.htaccess"
 PID_FILE="${INSTALL_DIR}/var/tt-rss.pid"
+HT_ACCESS_SECTION_DELIMITER="#Synology PHP"
+
+compute_generated_htaccess_section()
+{
+    local php_fpm="$1"
+    echo "${HT_ACCESS_SECTION_DELIMITER}
+AddHandler default-handler .htm .html .shtml
+AddHandler php-fastcgi .php
+AddType text/html .php
+Action php-fastcgi /${php_fpm}-handler.fcgi
+${HT_ACCESS_SECTION_DELIMITER}"
+}
 
+regenerate_htaccess()
+{
+    local php_fpm="$1"
+    if [ ! -f "${HT_ACCESS_FILE}" ]
+    then
+        echo "$(compute_generated_htaccess_section ${php_fpm})"
+        return 0
+    fi
+    local temp_file
+    local section
+    temp_file=$(mktemp --suffix=.htaccess tt-rss.XXXXX)
+    section=($(grep -n "${HT_ACCESS_SECTION_DELIMITER}" "${HT_ACCESS_FILE}" | cut -f 1 -d ':'))
+    if [ $? -ne 0 ]
+    then
+        echo "$(compute_generated_htaccess_section ${php_fpm})">"$temp_file"
+        cat "${HT_ACCESS_FILE}">>"$temp_file"
+    else
+        if [ "${#section[@]}" == "2" ]
+        then
+            local header_end
+            local trailer_start
+            header_end=$(expr ${section[0]} - 1)
+            trailer_start=$(expr ${section[1]} + 1)
+            head -n ${header_end} "${HT_ACCESS_FILE}" >"${temp_file}"
+            echo "$(compute_generated_htaccess_section ${php_fpm})" >>"${temp_file}"
+            tail -n +${trailer_start} "${HT_ACCESS_FILE}" >>"${temp_file}"
+        else
+            echo "$(compute_generated_htaccess_section ${php_fpm})">"${temp_file}"
+            cat "${HT_ACCESS_FILE}">>"$temp_file"
+        fi
+    fi
+    cat "${temp_file}"
+    rm "${temp_file}"
+    return 0
+}
 
 start_daemon ()
 {
-    start-stop-daemon -S -q -m -b -N 10 -x ${PHP} ${TTRSS} -c ${USER} -u ${USER} -p ${PID_FILE} \
-      -- --daemon 2> /dev/null
+    source "${INSTALL_DIR}/bin/php-settings.sh"
+    if [ ! -z "${php_fpm}" ]
+    then
+        echo "$(regenerate_htaccess ${php_fpm})" >"${HT_ACCESS_FILE}"
```

Comment:
There were several options here for this step:

1. Let the function `regenerate_htaccess` echo to the standard output the contents to put in the `.htaccess` file and perform the writing at this stage
2. Let the function `regenerate_htaccess` write "by itself" to the file

At this stage I chose the former option.